go/vt/vtgate/evalengine/testcases/cases.go

```diff
@@ -176,18 +178,67 @@ var Cases = []TestCase{
 	{Run: RegexpReplace},
 }
 
-func JSONPathOperations(yield Query) {
+func FnJSONKeys(yield Query) {
 	for _, obj := range inputJSONObjects {
 		yield(fmt.Sprintf("JSON_KEYS('%s')", obj), nil, false)
 
+		for _, path1 := range inputJSONPaths {
+			yield(fmt.Sprintf("JSON_KEYS('%s', '%s')", obj, path1), nil, false)
+		}
+	}
+}
+
+func FnJSONExtract(yield Query) {
+	for _, obj := range inputJSONObjects {
 		for _, path1 := range inputJSONPaths {
 			yield(fmt.Sprintf("JSON_EXTRACT('%s', '%s')", obj, path1), nil, false)
+
+			for _, path2 := range inputJSONPaths {
+				yield(fmt.Sprintf("JSON_EXTRACT('%s', '%s', '%s')", obj, path1, path2), nil, false)
+			}
+		}
+	}
+
+	yield(`JSON_EXTRACT('{"a": 1}', '$.a')`, nil, false)
+	yield(`JSON_EXTRACT('{"a": 1}', '$.*')`, nil, false)
+	yield(`JSON_EXTRACT('[1, 2, 3]', '$[0 to 2]')`, nil, false)
+	yield(`JSON_EXTRACT('{"a": 1, "b": 2}', '$.a', '$.b')`, nil, false)
+	yield(`JSON_EXTRACT('{"a": 1}', '$.a', '$.z')`, nil, false)
+
+	yield(`JSON_EXTRACT(CONCAT('{', '"a"', ':', ' ', '1', '}'), '$.a'))`, nil, false)
+	yield(`JSON_EXTRACT('{"a": 1}', CONCAT('$', '.', 'a'))`, nil, false)
+
+	yield(`JSON_EXTRACT(NULL, '$.a')`, nil, false)
+	yield(`JSON_EXTRACT(NULL, NULL)`, nil, false)
+
+	yield(`JSON_EXTRACT('{"a": 1}', NULL)`, nil, false)
+	yield(`JSON_EXTRACT('{"a": 1}', '$.a', NULL)`, nil, false)
+	yield(`JSON_EXTRACT('{"a": 1}', NULL, '$.a')`, nil, false)
+
+	yield(`JSON_EXTRACT('{"a": 1}', '$.b')`, nil, false)
+	yield(`JSON_EXTRACT('{"a": 1}', '$.b', '$.c')`, nil, false)
+	yield(`JSON_EXTRACT('[1,2,3]', '$[10]')`, nil, false)
+
+	yield(`JSON_EXTRACT('{invalid}', '$.a')`, nil, false)
+	yield(`JSON_EXTRACT('not json', '$.a')`, nil, false)
+	yield(`JSON_EXTRACT('', '$.a')`, nil, false)
+	yield(`JSON_EXTRACT('{invalid}', NULL)`, nil, false)
+
+	yield(`JSON_EXTRACT('{"a": 1}', '$.b[ 1 ].')`, nil, false)
+
+	yield(`JSON_EXTRACT(NULL, 'invalid-path')`, nil, false)
+	yield(`JSON_EXTRACT('{"a": 1}', NULL, 'invalid-path')`, nil, false)
+	yield(`JSON_EXTRACT('{"a": 1}', 'invalid-path', NULL)`, nil, false)
+	yield(`JSON_EXTRACT('{"a": 1}', '$.a', 'invalid')`, nil, false)
```

Comment:
Some of these could be handled by adding them to `inputJSONObjects` or `inputJSONPaths`, but that then highlights bugs in `JSON_KEYS` and `JSON_CONTAINS`, which I want to focus on in separate PRs.